chainstate/src/detail/error.rs

```diff
@@ -206,6 +206,16 @@ pub enum InitializationError {
     Block1Missing,
     #[error("Genesis mismatch: {0} according to configuration, {1} inferred from storage")]
     GenesisMismatch(Id<GenBlock>, Id<GenBlock>),
+    #[error(
+        "Node cannot load chainstate database because the versions mismatch: db `{0}`, app `{1}`"
+    )]
+    ChainstateStorageVersionMismatch(u32, u32),
+    #[error(
+        "Chain's config magic bytes do not match the one from database : expected `{0:?}`, actual `{1:?}`"
+    )]
+    ChainConfigMagicBytesMismatch([u8; 4], [u8; 4]),
+    #[error("Node's chain type doesn't match the one in the database : db `{0}`, app `{1}`")]
+    ChainTypeMismatch(String, String),
```

Comment:
These 3 errors can go into that module and become a single error and can be matched once